proxy/dkg.go

```diff
@@ -18,80 +18,95 @@ import (
 // NewDKG returns a new initialized DKG proxy
 func NewDKG(mngr txn.Manager, d dkgSrv.DKG, pk kyber.Point) DKG {
 	return dkg{
-		mngr: mngr,
-		d:    d,
-		pk:   pk,
+		manager:    mngr,
+		dkgService: d,
+		pk:         pk,
 	}
 }
 
 // dkg defines the DKG handlers
 //
 // - implements proxy.DKG
 type dkg struct {
-	mngr txn.Manager
-	d    dkgSrv.DKG
-	pk   kyber.Point
+	// manager - the transaction manager
+	manager txn.Manager
+	//dkgService - the DKG service
+	dkgService dkgSrv.DKG
+	// pk - the public key of the proxy
+	pk kyber.Point
 }
 
 // NewDKGActor implements proxy.DKG
+// Create a new DKG actor for the given electionID
 func (d dkg) NewDKGActor(w http.ResponseWriter, r *http.Request) {
 	var req types.NewDKGRequest
 
+	// Read the request
 	signed, err := types.NewSignedRequest(r.Body)
 	if err != nil {
 		InternalError(w, r, newSignedErr(err), nil)
 		return
 	}
 
+	// Verify the request
 	err = signed.GetAndVerify(d.pk, &req)
 	if err != nil {
 		InternalError(w, r, getSignedErr(err), nil)
 		return
 	}
 
+	// decode the electionID
 	electionIDBuf, err := hex.DecodeString(req.ElectionID)
 	if err != nil {
 		http.Error(w, "failed to decode electionID: "+req.ElectionID,
 			http.StatusBadRequest)
 		return
 	}
 
-	_, err = d.d.Listen(electionIDBuf, d.mngr)
+	// subscribe to the DKG service
+	_, err = d.dkgService.Listen(electionIDBuf, d.manager)
 	if err != nil {
 		http.Error(w, "failed to start actor: "+err.Error(),
 			http.StatusInternalServerError)
 		return
 	}
 }
 
+// Actor implements proxy.DKG
+// Send the actor status
 func (d dkg) Actor(w http.ResponseWriter, r *http.Request) {
 	w.Header().Set("Access-Control-Allow-Origin", "*")
 	w.Header().Set("Access-Control-Allow-Headers", "*")
 
 	vars := mux.Vars(r)
 
+	// check if electionID is present
 	if vars == nil || vars["electionID"] == "" {
 		http.Error(w, fmt.Sprintf("electionID not found: %v", vars), http.StatusInternalServerError)
 		return
 	}
 
 	electionID := vars["electionID"]
 
+	// decode the electionID
 	electionIDBuf, err := hex.DecodeString(electionID)
```

Comment:
This is an example of a line of code that is self-explenatory :)

Reply:
I was thinking that when you're reading the code,It may be good to have all the information by just reading the comments so you don't have to read every line of code to have an idea of how the function works